packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -87,7 +87,7 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
       });
       final HttpClientResponse response = await request.close();
       if (response.statusCode != HttpStatus.ok)
-        throw Exception('HTTP request failed, statusCode: ${response?.statusCode}, $resolved');
+        throw NetworkImageLoadException(code: response?.statusCode, uri: resolved);
```

Comment:
This should fail analysis because `code` is `@required`. 

But we already know that `response` is not null, so this should just be `response.statusCode` (and I realize this is a problem in the existing code :).